package/kedro_viz/data_access/managers.py

```diff
@@ -46,24 +47,17 @@ def __init__(self):
         self.registered_pipelines = RegisteredPipelinesRepository()
         self.tags = TagsRepository()
         self.modular_pipelines = ModularPipelinesRepository()
-        self._db_session = None
         # Make sure each registered pipeline has a distinct collection of edges.
         self.edges: Dict[str, GraphEdgesRepository] = defaultdict(GraphEdgesRepository)
         # Make sure the node dependencies are built separately for each registered pipeline.
         self.node_dependencies: Dict[str, Dict[str, Set]] = defaultdict(
             lambda: defaultdict(set)
         )
+        self.runs = RunsRepository()
 
-    @property
-    def db_session(self):  # pragma: no cover
-        """Sqlite db connection session"""
-        if not self._db_session:
-            logger.warning("Database connection was unsuccessful")
-        return self._db_session
-
-    @db_session.setter
-    def db_session(self, db_session: DatabaseSession):
-        self._db_session = db_session
+    def set_db_session(self, db_session_class: sessionmaker):
```

Comment:
Is `sessionmaker` really a type? I don't know much about sqlalchemy but just at a glance it looks weird not being camel case.

Reply:
Tbh I have no idea. Typing with sqlalchemy is very strange. I tried it both with `sessionmaker` and `Type[Session]` and both produce a different set of problems, so I'm parking it for now.